src/Compilers/CSharp/Portable/Binder/Binder_Lookup.cs

```diff
@@ -970,17 +972,22 @@ private static ImmutableArray<NamedTypeSymbol> GetBaseInterfaces(NamedTypeSymbol
 
             return result.ToImmutableAndFree();
 
-            static void addAllInterfaces(NamedTypeSymbol @interface, HashSet<NamedTypeSymbol> visited, ArrayBuilder<NamedTypeSymbol> result, ConsList<TypeSymbol> basesBeingResolved)
+            static void addAllInterfaces(NamedTypeSymbol @interface, HashSet<NamedTypeSymbol> visited, ArrayBuilder<NamedTypeSymbol> result, ConsList<TypeSymbol> basesBeingResolved, ImmutableHashSet<NamedTypeSymbol> cycleGuard)
             {
-                if (@interface.IsInterface && visited.Add(@interface))
+                if (@interface.IsInterface && !cycleGuard.Contains(@interface.OriginalDefinition) && visited.Add(@interface))
```

Comment:
>@interface.OriginalDefinition [](start = 67, length = 29)

Consider extracting a local for this value.

Reply:
> Consider extracting a local for this value.

I am not sure if it is worth doing. Implementations of OriginalDefinition are pretty inexpensive from allocation and perf point of view. However, capturing it in a local would lead to either introducing another nested ```if``` (after ```@interface.IsInterface``` check), or to use an inline assignment. 

---
In reply to: [325789306](https://github.com/dotnet/roslyn/pull/38738#discussion_r325789306) [](ancestors = 325789306)